### PR TITLE
Added person-video extraction

### PR DESCRIPTION
**Pull request by ldabas-msft**

### CODE REVIEW

File: agora/shopper_insights_api/src/video_processor.py [hunk 1]
```diff
@@ -26,8 +29,30 @@ def __init__(self, url, index, name, debug=False):
         self.inactivity_threshold = 30  # 30 seconds
```
Comment (fcabrera23) [inline agora/shopper_insights_api/src/video_processor.py:29]: myabe we should add a constructor flag that enables/disables rcording

File: agora/shopper_insights_api/src/video_processor.py [hunk 2]
```diff
@@ -26,8 +29,30 @@ def __init__(self, url, index, name, debug=False):
...
         self.debug = debug
 
+        # Initialize save_lock for thread-safe operations
+        self.save_lock = threading.Lock()
+
+        # Directory to store person images
+        #MNT PATH FOR ACSA SHOULD BE SET TO /app/detected_persons
```
Comment (fcabrera23) [inline agora/shopper_insights_api/src/video_processor.py:36]: Please talk with Seif/Mike to automate the setup of ACSA

File: agora/shopper_insights_api/src/video_processor.py [hunk 3]
```diff
@@ -26,8 +29,30 @@ def __init__(self, url, index, name, debug=False):
...
+            os.makedirs(self.person_image_dir)
+        
+        # Directory to store GIFs
+        self.videos_output_dir = "/app/detected_persons/videos"
+        if not os.path.exists(self.videos_output_dir):
+            os.makedirs(self.videos_output_dir)
+        
+        # Keep track of processed person directories
+        self.processed_person_dirs = set()
+
```
Comment (fcabrera23) [inline agora/shopper_insights_api/src/video_processor.py:48]: nit: extra line

File: agora/shopper_insights_api/src/video_processor.py [hunk 4]
```diff
@@ -26,8 +29,30 @@ def __init__(self, url, index, name, debug=False):
...
+            os.makedirs(self.videos_output_dir)
+        
+        # Keep track of processed person directories
+        self.processed_person_dirs = set()
+        
+
+        # Start the GIF creation thread
+        self.video_creation_thread = None
+
+
```
Comment (fcabrera23) [inline agora/shopper_insights_api/src/video_processor.py:53]: nit: extra line

File: agora/shopper_insights_api/src/video_processor.py [hunk 5]
```diff
@@ -26,8 +29,30 @@ def __init__(self, url, index, name, debug=False):
...
         # OpenVINO setup
-        MODEL_PATH = os.getenv("MODEL_PATH", "C:\\Users\\fcabrera\\Downloads\\jumpstart-apps\\agora\\contoso_hypermarket\\models")
+        MODEL_PATH = os.getenv("MODEL_PATH", "C:\\Users\\lakshitdabas\\Developer\\jumpstart-apps\\agora\\contoso_hypermarket\\models")
```
Comment (fcabrera23) [inline agora/shopper_insights_api/src/video_processor.py:55]: lets clean this and leave it to .\models as the default

File: agora/shopper_insights_api/src/video_processor.py [hunk 6]
```diff
@@ -76,11 +102,20 @@ def start(self):
             self.process_thread = threading.Thread(target=self.process_frames)
             self.process_thread.start()
             print(f"Started processing thread for video {self.index}")
+
+              # Start the video creation thread
+            if self.video_creation_thread is None or not self.video_creation_thread.is_alive():
```
Comment (fcabrera23) [inline agora/shopper_insights_api/src/video_processor.py:107]: add condition to start video thread if its enabeld per flag in constructor

File: agora/shopper_insights_api/src/video_processor.py [hunk 7]
```diff
@@ -135,6 +170,134 @@ def point_in_rectangle(self, point, rectangle):
         x, y = point
         x1, y1, x2, y2 = rectangle
         return x1 <= x <= x2 and y1 <= y <= y2
+    
+
```
Comment (fcabrera23) [inline agora/shopper_insights_api/src/video_processor.py:174]: nit: extra line

File: agora/shopper_insights_api/src/video_processor.py [hunk 8]
```diff
@@ -135,6 +170,134 @@ def point_in_rectangle(self, point, rectangle):
...
+            return
+
+        # Handle resampling filter compatibility
+        try:
+            resample_filter = Image.Resampling.LANCZOS  # Pillow >= 10.0.0
+        except AttributeError:
+            resample_filter = Image.LANCZOS  # Pillow < 10.0.0
+
+        frames = []
+        desired_size = (200, 400)  # Set desired width and height
```
Comment (fcabrera23) [inline agora/shopper_insights_api/src/video_processor.py:198]: why 200. 400? is this what we crop? What's the orignial video size?

File: agora/shopper_insights_api/src/video_processor.py [hunk 9]
```diff
@@ -173,19 +336,48 @@ def process_frames(self):
...
+                    #center = (center_x, center_y)
+
+                    # Check if the center point is within any detection area
+                    #is_in_area = any(
+                    #    self.point_in_rectangle(center, area) for area in self.restricted_areas
+                    #)
+
+                    # Save the detected person image only if they are in a detection area
+                    #if is_in_area:
+                    #    self.save_detected_person(frame, bbox)
```
Comment (fcabrera23) [inline agora/shopper_insights_api/src/video_processor.py:351]: REmove commentd code

File: agora/shopper_insights_api/src/video_processor.py [hunk 10]
```diff
@@ -213,6 +405,11 @@ def process_frames(self):
             self.current_shoppers = len(frame_shoppers)
             self.person_tracker = new_person_tracker
 
+            # Check for persons no longer being tracked to create videos
+            #for person_id in list(self.person_videos.keys()):
+            #    if person_id not in self.person_tracker:
+            #        self.create_person_video(person_id)
```
Comment (fcabrera23) [inline agora/shopper_insights_api/src/video_processor.py:411]: remove commented code

**Review by fcabrera23 — CHANGES_REQUESTED** [review]
Minor changes requeted @ldabas-msft

File: agora/shopper_insights_api/src/video_processor.py (hunk 8 above)
Comment (ldabas) [reply agora/shopper_insights_api/src/video_processor.py:198]: To make sure all frames have consistent dimensions, inconsistent frames causes errors when stitiching videos together

File: agora/shopper_insights_api/src/video_processor.py (hunk 1 above)
Comment (ldabas-msft) [reply agora/shopper_insights_api/src/video_processor.py:29]: added a flag in the constructor to do this

**Review by fcabrera23 — APPROVED** [review]
LGTM

**Pull request merged**